tests/components/template/test_cover.py

```diff
@@ -959,14 +1175,14 @@ async def test_set_tilt_position_optimistic(
     [
         (ConfigurationStyle.LEGACY, "icon_template"),
         (ConfigurationStyle.MODERN, "icon"),
+        (ConfigurationStyle.TRIGGER, "icon"),
     ],
 )
-async def test_icon_template(
-    hass: HomeAssistant, setup_single_attribute_state_cover
-) -> None:
+@pytest.mark.usefixtures("setup_single_attribute_state_cover")
+async def test_icon_template(hass: HomeAssistant) -> None:
     """Test icon template."""
     state = hass.states.get(TEST_ENTITY_ID)
-    assert state.attributes.get("icon") == ""
+    assert state.attributes.get("icon") in ("", None)
```

Comment:
Why is this changed? Same question for other tests testing attributes.

Reply:
Because trigger based entities don't execute the templates on startup so the attributes will be `None` however state based entities will resolve as an empty string.  This was done in other PRs.  I can come up with a different way to test this if needed.

Reply:
The added parametrization `initial_expected_state` looks good 👍 
Can you update also the other tests with the same pattern, `test_entity_picture_template`?